private/bufpkg/bufcheck/bufcheckserver/internal/buflintvalidate/field.go

```diff
@@ -1037,6 +1014,104 @@ func checkExampleValues(
 	return nil
 }
 
+// createSyntheticFieldDescriptorForChecking creates a clone of a specific field
+// into its own message so that it can be checked independently. For map fields,
+// this includes creating a synthetic MapEntry as well.
+func createSyntheticFieldDescriptorForChecking(
+	// If specified, parentMapFieldDescriptor is the map field to be cloned,
+	// e.g. it is a field on a message that points to a map type.
+	parentMapFieldDescriptor protoreflect.FieldDescriptor,
+	// If parentMapFieldDescriptor is specified, this is a MapKey or MapValue.
+	// Otherwise, it is the non-map field to be cloned.
+	fieldDescriptor protoreflect.FieldDescriptor,
+	resolver protodesc.Resolver,
+) (
+	newParentMapFieldDescriptor protoreflect.FieldDescriptor,
+	newFieldDescriptor protoreflect.FieldDescriptor,
+	err error,
+) {
+	// We need to determine the "root" descriptor: if parentMapFieldDescriptor
+	// is set, then fieldDescriptor points to a MapKey or MapValue field, but we
+	// want to create a synthetic clone of the map field itself.
+	rootFieldDescriptor := fieldDescriptor
+	if parentMapFieldDescriptor != nil {
+		rootFieldDescriptor = parentMapFieldDescriptor
+	}
+	originalFile := rootFieldDescriptor.ContainingMessage().ParentFile()
+	newFileDescriptorProto := &descriptorpb.FileDescriptorProto{
+		Name:    proto.String(":synthetic:"),
+		Package: proto.String("__SYNTHETIC__"),
+		Syntax:  proto.String("proto3"),
+		Dependency: []string{
+			originalFile.Path(),
+		},
+	}
+
+	// Ensure that enum and message types are available to be resolved. It's a
+	// bit tricky to get this right, so we just import everything the original
+	// file containing the root message imports (and that file, too.)
+	for i := 0; i < originalFile.Imports().Len(); i++ {
+		newFileDescriptorProto.Dependency = append(
+			newFileDescriptorProto.Dependency,
+			originalFile.Imports().Get(i).Path(),
+		)
+	}
```

Comment:
If you just mark the one import above (the original file) as a public import, you don't need any of this.

Reply:
Er... never mind, I'm wrong. Ignore this comment. You'd need to instead actually _edit_ the original file to make all of _its_ imports public, which you definitely don't want to hassle with...